Source/WebCore/platform/graphics/gstreamer/mse/MediaPlayerPrivateGStreamerMSE.cpp

```diff
@@ -218,6 +218,23 @@ void MediaPlayerPrivateGStreamerMSE::pause()
     updateStates();
 }
 
+bool MediaPlayerPrivateGStreamerMSE::paused() const
```

Comment:
paused() implementation from MediaPlayerPrivateMSE class differs only by not including m_isPlaybackRatePaused. Why we need that difference in MSE path?

Reply:
That's because MediaPlayerPrivateGStreamerMSE in its play()/pause() function does no longer calls parent class implementations - MediaPlayerPrivateGStreamer::play() and MediaPlayerPrivateGStreamer::pause().
This was changed in this commit https://github.com/WebKit/WebKit/commit/9382c6ef1a5b64d1701c691c63fe84de22cf932b.
These functions in MediaPlayerPrivateGStreamer class are the only ones that set m_isPlaybackRatePaused variable. If that is not happening anymore when using MediaPlayerPrivateGStreamerMSE then also MediaPlayerPrivateGStreamerMSE should not use paused() implementation from parent class that reads m_isPlaybackRatePaused. The value of that variable will simply be invalid. Hence the new implementation of MediaPlayerPrivateGStreamerMSE::paused() was added that doesn't use  m_isPlaybackRatePaused.